hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1691,24 +1682,39 @@ func reconcileCAPICluster(cluster *capiv1.Cluster, hcluster *hyperv1.HostedClust
 	cluster.Annotations = map[string]string{
 		hostedClusterAnnotation: client.ObjectKeyFromObject(hcluster).String(),
 	}
-	gvk, err := apiutil.GVKForObject(infraCR, api.Scheme)
-	if err != nil {
-		return err
-	}
-	cluster.Spec = capiv1.ClusterSpec{
-		ControlPlaneEndpoint: capiv1.APIEndpoint{},
-		ControlPlaneRef: &corev1.ObjectReference{
-			APIVersion: "hypershift.openshift.io/v1alpha1",
-			Kind:       "HostedControlPlane",
-			Namespace:  hcp.Namespace,
-			Name:       hcp.Name,
-		},
-		InfrastructureRef: &corev1.ObjectReference{
-			APIVersion: gvk.GroupVersion().String(),
-			Kind:       gvk.Kind,
-			Namespace:  infraCR.GetNamespace(),
-			Name:       infraCR.GetName(),
-		},
+	var gvk schema.GroupVersionKind
+	var err error
+	if infraCR != nil { //non platform none
```

Comment:
nit: let's be consistent with the rest of the code base and use block comment style. Inlining here does to seems to add value.